CONTRIBUTING.rst

```diff
@@ -51,22 +51,63 @@ This section include some tips that will help you to push source code.
 Dependencies
 ~~~~~~~~~~~~
 
-Needed libraries should be installed in this way:
+Our build system is based on CMake, so we need to have `CMake 3.5 or higher <https://cmake.org/>`_
+installed. As we will deal with languages that build native binaries, we will
+need to have installed any of the `supported CMake build tools <https://cmake.org/cmake/help/v3.5/manual/cmake-generators.7.html>`_.
+
+On Linux and Mac, we recommend installing GNU g++ 6.1 or higher, on Windows
+we only support `MinGW64 <http://mingw-w64.org>`_ at the moment.
+Note that a prerequiste for the C++ toolchain is that C++14 must be supported.
+
+For the python code, we need some libraries that can be installed in this way:
 
 .. code:: sh
 
     # Depending on the system and setup to append "sudo -H" before could be needed.
     pip3 install -r requirements-dev.txt
 
+Building
+~~~~~~~~
+
+The preferred way CMake is meant to be used, is by setting up an "out of source" build.
+So in order to build our native code, we have to follow these steps:
+
+Linux and Mac
+
+.. code::
+
+    qiskit-sdk-py$ mkdir out
+    qiskit-sdk-py$ cd out
+    qiskit-sdk-py/out$ cmake ..
+    qiskit-sdk-py/out$ make
+
+Windows
+
+.. code::
+
+    C:\..\> mkdir out
+    C:\..\> cd out
+    C:\..\out> cmake -DUSER_LIB_PATH=C:\path\to\mingw64\lib\libpthreads.a -G "MinGW Makefiles" ..
+    C:\..\out> make
+
+As you can see, the Windows cmake command invocation is slightly different from
+the Linux and Mac version, this is because we need to provide CMake with some
+more info about where to find libphreads.a for later building. Furthermore,
+we are forcing CMake to generate MingGW makefiles, because we don't support
+other toolchain at the moment.
+
+
 Test
 ~~~~
 
 New features often imply changes in the existent tests or new ones are
 needed. Once they're updated/added run this be sure they keep passing:
 
+All platforms:
 .. code:: sh
+    $> cd out
+    out$> ctest -VV
```

Comment:
For consistency, can we also update https://github.com/QISKit/qiskit-sdk-py/blob/master/doc/dev_introduction.rst#testing, as it refers to the previous `make test`? The passing of environment variables seem to work in the same way.

Also, if there was a way of making `-VV` the default verbosity just fot the `test` target (the one generated by cmake), maybe it would be more discoverable to  `out$ make test` instead of `ctest`?

I cannot make an inline comment on that line, but can we also revise the commands on the `Style guide` section further down the file and also add `make style`?

Reply:
I haven't found a way to run '`make test` target generated by CMake without explicitly passing some arguments to the command. I'll update the documentation to use `make test` with some more arguments and keep coherence  though.

> I cannot make an inline comment on that line, but can we also revise the commands on the Style 
> guide section further down the file and also add make style?

I guess you mean: `make linter` right?  ... which BTW sounds not very standard... I'll change it to: ` make lint`.

Reply:
Nice! Keeping coherence is indeed the main goal!

> I guess you mean: make linter right? ... which BTW sounds not very standard... I'll change it to: make lint.

I actually meant explicitly mentioning _both_  (`style` and `linter`) in the document, as currently there is a bit of a mismatch: Travis runs both targets for the linting job, and ideally contributors should as well be executing both (but we are not mentioning `style` anywhere). Renaming `linter` to `lint` makes sense - and even merging both current targets into one would also probably be a good thing!

Reply:
Ok, I'll update the doc to add the `make style` step.
> ... and even merging both current targets into one would also probably be a good thing!

yeah, could make sense keep them together, if it's ok with you, I'm gonna keep them separated for now, because I want to merge this PR asap... but let's put the radar on this and discuss this for the next iteration.